### PR TITLE
(WIP) Separate ArenaConfigState from GameConfig fields associated with arena.

### PR DESCRIPTION
**Pull request by Unengine**
## Separated variables in `GameConfig`
- ArenaScoreDefault (int)
- ArenaChallengeCountMax (int)

## Migrated variables in `GameConfigState`
- DailyRewardInterval
- WeeklyArenaInterval

### CODE REVIEW

**Review by ipdae — COMMENTED** [review]
this is breaking change in genesis. doesn't it matter?

**Review by moreal — APPROVED** [review]

**Review by moreal — CHANGES_REQUESTED** [review]
The test seems failed 😿 

```
  X Lib9c.Tests.Action.HackAndSlashTest.Execute(avatarLevel: 100, worldId: 1, stageId: 25, contains: True) [202ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at Nekoyume.Model.State.ArenaInfo..ctor(AvatarState avatarState, ArenaConfigState arenaConfigState, CharacterSheet characterSheet, Boolean active) in /home/runner/work/lib9c/lib9c/Lib9c/Model/State/WeeklyArenaState.cs:line 362
   at Nekoyume.Model.State.WeeklyArenaState.Update(AvatarState avatarState, ArenaConfigState arenaConfigState, CharacterSheet characterSheet, Boolean active) in /home/runner/work/lib9c/lib9c/Lib9c/Model/State/WeeklyArenaState.cs:line 166
   at Nekoyume.Model.State.WeeklyArenaState.Set(AvatarState avatarState, ArenaConfigState arenaConfigState, CharacterSheet characterSheet) in /home/runner/work/lib9c/lib9c/Lib9c/Model/State/WeeklyArenaState.cs:line 176
   at Nekoyume.Action.HackAndSlash.Execute(IActionContext context) in /home/runner/work/lib9c/lib9c/Lib9c/Action/HackAndSlash.cs:line 259
   at Lib9c.Tests.Action.HackAndSlashTest.Execute(Int32 avatarLevel, Int32 worldId, Int32 stageId, Boolean contains) in /home/runner/work/lib9c/lib9c/.Lib9c.Tests/Action/HackAndSlashTest.cs:line 108
```

**Comment by Unengine** [comment]
> this is breaking change in genesis. doesn't it matter?

I think it can be matter. I'd rather not treat this as a state, but an non-serialized class using only csv data.

**Comment by ipdae** [comment]
conflict submodule

**Review by boscohyun — COMMENTED** [review]
It would be nice if basic serialization unit tests were added for the newly added ʻArenaConfig` model.

File: Lib9c/Model/Config/ArenaConfig.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            foreach (var row in sheet.Values)
+            {
+                Update(row);
+            }
+        }
+
+        public void Set(ArenaConfigSheet sheet)
+        {
+            foreach (var row in sheet)
```
Comment (limebell) [inline Lib9c/Model/Config/ArenaConfig.cs:36]: Is this okay to be used without explicit ordering?

Comment (boscohyun) [reply Lib9c/Model/Config/ArenaConfig.cs:36]: It seems like you can use `sheet.OrderedList`.

Comment (Unengine) [reply Lib9c/Model/Config/ArenaConfig.cs:36]: Order of the elements is not important in this situation.

File: Lib9c/Model/Config/IConfig.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+
+namespace Nekoyume.Model.Config
+{
+    public interface IConfig
+    {
+    }
```
[the remark this answers is not in the document]
Comment (Unengine) [reply Lib9c/Model/Config/IConfig.cs:6]: It is intended to use it as a marker interface and it's used in `FailedToInstantiateConfigException<T> : Exception where T : IConfig` in game project.

**Comment by moreal** [comment]
If it will not be processed anymore, how about closing this pull request? @Unengine

**Pull request closed without merging**